UnitTests/GitUITests/CommandsDialogs/FormCommitTests.cs

```diff
@@ -304,28 +304,52 @@ private void RunFormTest(Action<FormCommit> testDriver, CommitKind commitKind =
 
         private void RunFormTest(Func<FormCommit, Task> testDriverAsync, CommitKind commitKind = CommitKind.Normal)
         {
-            UITest.RunForm(
-                () =>
-                {
-                    switch (commitKind)
+            using (var mainForm = new Form())
+            {
+                UITest.RunForm<FormCommit>(
+                    () =>
                     {
-                        case CommitKind.Normal:
-                            Assert.True(_commands.StartCommitDialog(owner: null));
-                            break;
-
-                        case CommitKind.Squash:
-                            Assert.True(_commands.StartSquashCommitDialog(owner: null, _referenceRepository.Module.GetRevision()));
-                            break;
-
-                        case CommitKind.Fixup:
-                            Assert.True(_commands.StartFixupCommitDialog(owner: null, _referenceRepository.Module.GetRevision()));
-                            break;
-
-                        default:
-                            throw new ArgumentException($"Unsupported commit kind: {commitKind}", nameof(commitKind));
-                    }
-                },
-                testDriverAsync);
+                        mainForm.Shown += (s, e) =>
+                        {
+                            switch (commitKind)
+                            {
+                                case CommitKind.Normal:
+                                    Assert.True(_commands.StartCommitDialog(owner: mainForm));
+                                    break;
+
+                                case CommitKind.Squash:
+                                    Assert.True(_commands.StartSquashCommitDialog(owner: mainForm, _referenceRepository.Module.GetRevision()));
+                                    break;
+
+                                case CommitKind.Fixup:
+                                    Assert.True(_commands.StartFixupCommitDialog(owner: mainForm, _referenceRepository.Module.GetRevision()));
+                                    break;
+
+                                default:
+                                    throw new ArgumentException($"Unsupported commit kind: {commitKind}", nameof(commitKind));
+                            }
+                        };
+                        if (Application.MessageLoop)
+                        {
+                            mainForm.Show(owner: null);
+                        }
+                        else
+                        {
+                            Application.Run(mainForm);
```

Comment:
1. Does `form.ShowDialog(owner)` run a message loop as `Application.Run` does? I guess not.
(When running the tests, `Application.MessageLoop` is `false`.)
With this surrounding `mainForm`, I see no `OperationCanceledException`s from the regarding tests anymore.
But some `FormCommitTests` fail because `WaitForPendingOperations` does not suffice.
2. What is special about async loading of FormCommit?

Reply:
If `FormCommit` started on its own (e.g. from a Windows Explorer context menu) then there is no application loop, and the form is the entry form for the application.
If the form is started from `FormBrowse` we already have the message loop.

Reply:
I'm not sure why we have this code here though

Reply:
> I'm not sure why we have this code here though

Because I copied it from `StartBrowseDialog` as an experiment. It seems to have special handling for the test execution.
I suppose that the `CancellationToken` is canceled when the message loop ends unexpectedly.

I know `Form.ShowDialog` runs an own message loop because the form shall be modal.
My question is:  Is this message loop really compatible to the test runner environment?
Or is a surrounding dummy main form with the message loop from `Application.Run(form)` required?